components/application-mgt/org.wso2.carbon.device.application.mgt.core/src/main/java/org/wso2/carbon/device/application/mgt/core/impl/SubscriptionManagerImpl.java

```diff
@@ -52,59 +52,17 @@
-                    .common.DeviceIdentifier(device.getId(), device.getType());
-            try {
-                DeviceManagementProviderService dmpService = DataHolder.getInstance().getDeviceManagementService();
-                if (!dmpService.isEnrolled(deviceIdentifier)) {
-                    log.error("Device with ID: " + device.getId() + " is not enrolled to install the application.");
-                } else {
-                    if (log.isDebugEnabled()) {
-                        log.debug("Installing application to : " + device.getId());
-                    }
-                    //Todo: generating one time download link for the application and put install operation to device.
-
-                    // put app install operation to the device
-                    ProfileOperation operation = new ProfileOperation();
-                    operation.setCode("INSTALL_APPLICATION");
-                    operation.setType(Operation.Type.PROFILE);
-                    operation.setPayLoad("{'type':'enterprise', 'url':'http://10.100.5.76:8000/app-debug.apk', 'app':'" + applicationUUID + "'}");
-                    List<org.wso2.carbon.device.mgt.common.DeviceIdentifier> deviceIdentifiers = new ArrayList<>();
-                    deviceIdentifiers.add(deviceIdentifier);
-                    DeviceManagementProviderService deviceManagementProviderService = HelperUtil.getDeviceManagementProviderService();
-                    deviceManagementProviderService.addOperation(DeviceManagementConstants.MobileDeviceTypes.MOBILE_DEVICE_TYPE_ANDROID,
-                            operation, deviceIdentifiers);
-
-                    DeviceApplicationMapping deviceApp = new DeviceApplicationMapping();
-                    deviceApp.setDeviceIdentifier(device.getId());
-                    deviceApp.setApplicationUUID(applicationUUID);
-                    deviceApp.setVersionName(versionName);
-                    deviceApp.setInstalled(false);
-                    deviceManagementProviderService.addDeviceApplicationMapping(deviceApp);
-//                    DAOFactory.getSubscriptionDAO().addDeviceApplicationMapping(device.getId(), applicationUUID, false);
-                    failedDeviceList.remove(device);
-                }
-            } catch (DeviceManagementException | OperationManagementException | InvalidDeviceException e) {
-                throw new ApplicationManagementException("Failed to install application " + applicationUUID + " on device " + deviceIdentifier, e);
-            }
-        }
-        return failedDeviceList;
+        return installApplication(applicationUUID, deviceList, versionName);
     }
 
     @Override
-    public List<DeviceIdentifier> installApplicationForUsers(String applicationUUID, List<String> userList)
+    public List<DeviceIdentifier> installApplicationForUsers(String applicationUUID, List<String> userList, String versionName)
```

Comment:
Seems to be going beyond 120 line limit. In the following methods , shall we fix them?

Reply:
+1